src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -34,4 +39,116 @@ public function __construct(ContainerBuilder $container, FileLocatorInterface $l
 
         parent::__construct($locator);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setCurrentDir($dir)
+    {
+        $this->currentDir = $dir;
+        parent::setCurrentDir($dir);
+    }
+
+    public function registerClasses(Definition $prototype, $psr4Glob, $directory)
+    {
+        $directory = $this->locator->locate($directory, $this->currentDir, true);
+        $directory = realpath($directory) ?: $directory;
+        if (!$classes = $this->findClasses($psr4Glob, $directory)) {
+            throw new InvalidArgumentException(sprintf('No classes matching "%s" were found in directory "%s".', $psr4Glob, $directory));
+        }
+
+        foreach ($classes as $class) {
+            $this->container->setDefinition($class, clone $prototype);
+        }
+    }
+
+    private function findClasses($psr4Glob, $directory)
+    {
+        $missingClassException = new \ReflectionException();
+        $throwingAutoloader = function () use ($missingClassException) { throw $missingClassException; };
+        spl_autoload_register($throwingAutoloader);
+        $classes = array();
+
+        try {
+            if (!preg_match('/^(?:[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+\\\\)++/', $psr4Glob, $match)) {
+                throw new InvalidArgumentException(sprintf('Namespace glob does not start with a valid PSR-4 prefix: %s.', $psr4Glob));
+            }
+
+            $psr4Prefix = $match[0];
+            $classRegexp = self::patternToRegexp($psr4Glob);
+            $excludeTest = false === strpos($psr4Glob, 'Test', strlen($psr4Prefix));
```

Comment:
I think we do, because it gives a good default: test folder/classes usually contain strange things that can break PHP (look at our own test suite - this is were I needed this default first).